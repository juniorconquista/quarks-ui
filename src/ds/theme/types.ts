import {
    ColorProps,
    FlexboxProps,
    LayoutProps,
    SpaceProps,
    PositionProps,
    BorderProps,
    GridProps,
} from 'styled-system';

export type DefaultStyleProps = BorderProps &
    ColorProps &
    FlexboxProps &
    LayoutProps &
    PositionProps &
    SpaceProps &
    GridProps & {
        // for some reason this isn't included; this is for replacing the 'base' component
        // as?: React.ComponentType;
    };
