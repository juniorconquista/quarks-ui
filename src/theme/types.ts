import {
    ColorProps,
    FlexboxProps,
    LayoutProps,
    SpaceProps,
    PositionProps,
    BorderProps,
    GridProps,
} from 'styled-system';

export type CssProperties = {
    transition?: string;
    filter?: string;
    textTransform?:
        | 'none'
        | 'capitalize'
        | 'uppercase'
        | 'lowercase'
        | 'initial'
        | 'inherit';
    animationFillMode?:
        | 'none'
        | 'forwards'
        | 'backwards'
        | 'both'
        | 'initial'
        | 'inherit';
    objectFit?: 'contain' | 'cover' | 'fill' | 'none' | 'scale-down';
    whiteSpace?:
        | 'none'
        | 'nowrap'
        | 'pre'
        | 'pre-line'
        | 'pre-wrap'
        | 'initial'
        | 'inherit';
    '-webkit-box-shadow'?: string;
};

export type DefaultStyleProps = BorderProps &
    ColorProps &
    FlexboxProps &
    LayoutProps &
    PositionProps &
    SpaceProps &
    GridProps & {
        // for some reason this isn't included; this is for replacing the 'base' component
        // as?: React.ComponentType;
    };
