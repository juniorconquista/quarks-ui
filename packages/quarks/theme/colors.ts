
// teste
export default (dark = false): any => ({
  dark,
  get primary () {
    return {
      default: this.dark ? '#375a7f' : '#007bff',
      contrast: this.dark ? '#2f75bf' : '#6aa5e4',
      dark: '#375a7f',
      light: '#007bff'
    }
  },
  get secondary () {
    return {
      default: this.dark ? '#444444' : '#6c757d',
      dark: '#444444',
      light: '#6c757d'
    }
  },
  get success () {
    return {
      default: this.dark ? '#00bc8c' : '#28a745',
      dark: '#00bc8c',
      light: '#28a745'
    }
  },
  get danger () {
    return {
      default: this.dark ? '#e74c3c' : '#dc3545',
      dark: '#e74c3c',
      light: '#dc3545'
    }
  },
  get warning () {
    return {
      default: this.dark ? '#f39c12' : '#ffc107',
      dark: '#f39c12',
      light: '#ffc107'
    }
  },
  get info () {
    return {
      default: this.dark ? '#3498db' : '#17a2b8',
      dark: '#3498db',
      light: '#17a2b8'
    }
  },
  get background () {
    return {
      default: this.dark ? '#f8f9fa' : '#343a40',
      dark: '#f8f9fa',
      light: '#343a40'
    }
  },
  get text () {
    return {
      default: this.dark ? '#f8f9fa' : '#343a40',
      dark: '#f8f9fa',
      light: '#343a40'
    }
  }
})
