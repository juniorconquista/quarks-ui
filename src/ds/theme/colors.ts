export default {
    dark: false,
    primary: {
        default: '#F26522',
        dark: '#d14b0c',
        light: '#f69263',
        contrast: '#ffffff',
    },
    secondary: {
        default: '#372E2A',
        dark: '#2c2522',
        light: '#453935',
        contrast: '#1B2B3A',
    },
    danger: {
        default: '#ff6961',
        dark: '#ff261b',
        light: '#ffbdb9',
        contrast: '#ffffff',
    },
    success: {
        default: '#0E9d51',
        dark: '#0a753c',
        light: '#10bc60',
        contrast: '#ffffff',
    },
    warning: {
        default: '#ffb143',
        dark: '#e4902c',
        light: '#ffd868',
        contrast: '#ffffff',
    },
    info: {
        default: '#0ea7a7',
        dark: '#0a7676',
        light: '#11d4d4',
        contrast: '#ffffff',
    },
    grays: {
        default: '#9ba8b2',
        dark: '#4b4b4b',
        light: '#eeeeef',
        contrast: '#f5f5f5',
    },
    background: {
        default: '#ffffff',
        dark: '#F26522',
    },
    text: {
        light: '#ffffff',
        dark: '#1B2b3a',
    },
    disabled: '#9398a6',
    primaries: ['#d7edf4'],
    whites: ['#eeeeef', '#f2fbfe'],
    blacks: ['#35363a'],
    hovers: ['#f5f7f7'],
};
