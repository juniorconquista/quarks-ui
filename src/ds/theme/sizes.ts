export const breakpoints = ['0rem', '37.5rem', '62rem'];

export const space = {
    none: '0',
    xxxs: '0.285rem',
    xxs: '0.571rem',
    xs: '0.857rem',
    sm: '1.142rem',
    md: '1.714rem',
    lg: '2.5rem',
    xl: '3.428rem',
    xxl: '6.857rem',
};

export const fontSizes = [
    '0.5rem',
    '0.85rem',
    '1rem',
    '1.5rem',
    '1.8rem',
    '2rem',
];

export const buttonSizes = {
    medium: {
        paddingTop: 8,
        paddingBottom: 8,
        paddingRight: 16,
        paddingLeft: 16,
        borderRadius: 16,
    },
    small: {
        paddingTop: 4,
        paddingBottom: 4,
        paddingRight: 16,
        paddingLeft: 16,
        borderRadius: 16,
    },
};

export const radii = {
    none: '0',
    sm: '0.285rem',
    md: '2.25rem',
    lg: '1.142rem',
    xl: '1.714rem',
};
