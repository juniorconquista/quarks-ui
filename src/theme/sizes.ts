export const breakpoints = ['0rem', '37.5rem', '62rem']

export const space = {
  none: '0',
  xxxs: '0.285rem',
  xxs: '0.571rem',
  xs: '0.857rem',
  sm: '1.142rem',
  md: '1.714rem',
  lg: '2.5rem',
  xl: '3.428rem',
  xxl: '6.857rem'
}

export const fontSizes = [
  '0.5rem', // 8px
  '0.67rem', // 10.72px - H6 / SMALL
  '0.83rem', // 13.28px - H5
  '0.875rem', // 14px
  '1rem', // 16px - H4/P/SPAN
  '1.17rem', // 18.72px - H3
  '1.5rem', // 24px - H2
  '1.8rem', // 28.8px
  '2rem' // 32px - H1
]

export const buttonSizes = {
  medium: {
    paddingTop: 8,
    paddingBottom: 8,
    paddingRight: 16,
    paddingLeft: 16,
    borderRadius: 16
  },
  small: {
    paddingTop: 4,
    paddingBottom: 4,
    paddingRight: 16,
    paddingLeft: 16,
    borderRadius: 16
  }
}

export const radii = {
  none: '0',
  sm: '0.285rem',
  md: '2.25rem',
  lg: '1.142rem',
  xl: '1.714rem'
}
