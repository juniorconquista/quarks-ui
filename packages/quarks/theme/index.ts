import colors from './colors'
import { breakpoints, fontSizes, fontWeights, radii, space } from './sizes'

const theme = {
  breakpoints,
  fontSizes,
  fontWeights,
  radii,
  space,
  colors: colors()
}

// @ts-expect-error
theme.breakpoints.mobile = breakpoints[0]
// @ts-expect-error
theme.breakpoints.tablet = breakpoints[1]
// @ts-expect-error
theme.breakpoints.desktop = breakpoints[2]

export default theme
