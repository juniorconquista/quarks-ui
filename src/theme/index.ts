import colors from './colors';
import { breakpoints, space, fontSizes, radii } from './sizes';

const theme = {
    fontSizes,
    space,
    breakpoints,
    radii,
    colors: colors(),
};

//@ts-ignore
theme.breakpoints.mobile = breakpoints[0];
//@ts-ignore
theme.breakpoints.tablet = breakpoints[1];
//@ts-ignore
theme.breakpoints.desktop = breakpoints[2];

export default theme;
