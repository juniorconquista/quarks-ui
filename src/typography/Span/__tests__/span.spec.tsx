import React from 'react'
import renderer from 'react-test-renderer'
import ThemeTest from '../../../utils/ThemeProviderTest'
import 'jest-styled-components'

import SPan from '../SPan'
import theme from '../../../theme'

describe('SPan', () => {
  test('basic snapshot', () => {
    const span = renderer
      .create(
                <ThemeTest>
                    <SPan styling="base" />
                </ThemeTest>
      )
      .toJSON()
    expect(span).toMatchSnapshot()
  })

  test('should inherit the styling properties of the base variant', () => {
    const span = renderer
      .create(
                <ThemeTest>
                    <SPan styling="base" />
                </ThemeTest>
      )
      .toJSON()
    expect(span).toHaveStyleRule('color', theme.colors.text.default)
    expect(span).toHaveStyleRule('font-size', theme.fontSizes[4])
  })

  test('should apply custom styling properties', () => {
    const span = renderer
      .create(<SPan styling="base" mt="20px" p="3px" />)
      .toJSON()
    expect(span).toHaveStyleRule('margin-top', '20px')
    expect(span).toHaveStyleRule('padding', '3px')
  })
})
