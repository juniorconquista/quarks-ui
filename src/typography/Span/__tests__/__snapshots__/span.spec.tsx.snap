// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SPan basic snapshot 1`] = `
.c0 {
  color: #343a40;
  font-size: 1rem;
}

<span
  className="c0"
/>
`;
